R/mergingandstackingutilities.R

```diff
@@ -567,6 +553,14 @@ sanitizeSPSSVariableNames <- function(variable.names) {
                                              FUN.VALUE = character(1))
     }
 
+    # Can't begin with or end with a period
+    starts.or.ends.with.period <- startsWith(variable.names, ".") | endsWith(variable.names, ".")
+    if (any(starts.or.ends.with.period)) {
+        warning("Cannot save variables names which begin or end with '.'. Some variables have had '.' removed from their names: ",
+                paste0(variable.names[forbidden.period], collapse = ", "))
+        variable.names[starts.or.ends.with.period] <- gsub("^\\.+", "", gsub("\\.+$", "", variable.names[starts.or.ends.with.period]))
+    }
+
```

Comment:
Why did you change the order? What if they are reserved keywords after you've removed the '.'s?

Reply:
Good point. I guess I didn't want variable names to end with a period after trimming to be under 64 bytes. I'll have another look at this.